airbyte-webapp/src/components/connection/CatalogTree/CatalogSection.tsx

```diff
@@ -96,6 +98,32 @@ const CatalogSectionInner: React.FC<CatalogSectionInnerProps> = ({
     [updateStreamWithConfig]
   );
 
+  const numberOfFieldsInStream = Object.keys(streamNode?.stream?.jsonSchema?.properties).length ?? 0;
+
+  const onSelectedFieldsUpdate = (selectedFields: SelectedFieldInfo[]) => {
```

Comment:
A thought: if `onSelectedFieldsUpdate` took `previouslySelectedFields` as a second param, instead of just `selectedFields`, then the special-case props `onFirstFieldDeselected` and `onAllFieldsSelected` could go away and and the fact that those are special cases would be more cleanly scoped to this one file.

Reply:
Good point, I think that would indeed be cleaner! As discussed offline, I'll defer this to a follow-up PR